File: smells/exec.js
const vscode = require('vscode');

var smell = {

    detect : (token) => {

        if(token.hasOwnProperty("line")) var lineno = token.line;
        if(token.hasOwnProperty("type")) var tokenType = token.type;
        if(token.hasOwnProperty("name")) var name= token.name;
        if(token.hasOwnProperty("args")) var args= token.args;
        if(token.hasOwnProperty("hasInputs")) var hasInputs= token.hasInputs;
        
        if(tokenType == "function_call" && name == "exec" && args!=null && hasInputs == "True"){

            const warning = 'possible exec statement at line '+ lineno;
            vscode.window.showWarningMessage(warning);
        }
    }
}

module.exports = smell;

File: py/smells/exec.py
from operations.savewarnings import saveWarnings

def detect(token) :

    if token.__contains__("line"): lineno = token["line"]
    if token.__contains__("type"): tokenType = token["type"] 
    if token.__contains__("name"): methodname = token["name"]
    if token.__contains__("args"): args = token["args"]
    if token.__contains__("hasInputs"): containsUserInput =  token["hasInputs"]

    if (tokenType == "function_call" and methodname == "exec" and args and containsUserInput):
        warning = 'exec statement'
        saveWarnings(warning,str(lineno))
        print(warning+ ' at line '+ str(lineno))
